lib/lightstep/tracer.rb

```diff
@@ -183,28 +183,38 @@ def configure(component_name:, access_token: nil, transport: nil, tags: {})
     CARRIER_TRACER_STATE_PREFIX = 'ot-tracer-'.freeze
     CARRIER_BAGGAGE_PREFIX = 'ot-baggage-'.freeze
 
+    CARRIER_SPAN_ID = (CARRIER_TRACER_STATE_PREFIX + 'spanid').freeze
+    CARRIER_TRACE_ID = (CARRIER_TRACER_STATE_PREFIX + 'traceid').freeze
+    CARRIER_SAMPLED = (CARRIER_TRACER_STATE_PREFIX + 'sampled').freeze
+
     DEFAULT_MAX_LOG_RECORDS = 1000
     MIN_MAX_LOG_RECORDS = 1
     DEFAULT_MAX_SPAN_RECORDS = 1000
     MIN_MAX_SPAN_RECORDS = 1
 
     def inject_to_text_map(span, carrier)
-      carrier[CARRIER_TRACER_STATE_PREFIX + 'spanid'] = span.span_context.id
-      carrier[CARRIER_TRACER_STATE_PREFIX + 'traceid'] = span.span_context.trace_id unless span.span_context.trace_id.nil?
-      carrier[CARRIER_TRACER_STATE_PREFIX + 'sampled'] = 'true'
+      carrier[CARRIER_SPAN_ID] = span.span_context.id
+      carrier[CARRIER_TRACE_ID] = span.span_context.trace_id unless span.span_context.trace_id.nil?
+      carrier[CARRIER_SAMPLED] = 'true'
 
       span.span_context.baggage.each do |key, value|
         carrier[CARRIER_BAGGAGE_PREFIX + key] = value
       end
     end
 
     def extract_from_text_map(operation_name, carrier)
+      # If the carrier does not have both the span_id and trace_id key
+      # skip the processing and just return a normal span
+      if !carrier.has_key?(CARRIER_SPAN_ID) || !carrier.has_key?(CARRIER_TRACE_ID)
```

Comment:
This reads much better for me, too. in my brief travels in Ruby-land, though, it seems like whatever makes the most sense to me is wrong :)

Thankfully you seem to be on my side, and I'm happy to merge it this way instead. Thanks.